util.go

```diff
@@ -46,12 +47,31 @@ func strListContains(haystack []string, needle string) bool {
 	return false
 }
 
+const ossVault = `15cd22ce-24af-43a4-aa83-4c1a36a4b177`
+const entVault = `b2c13ec1-60e8-4733-9a76-88dbb2ce2471`
+
 // userAgent determines the User Agent to send on HTTP requests. This is mostly copied
 // from the useragent helper in vault and may get replaced with something more general
 // for plugins
 func userAgent() string {
-	version := os.Getenv(pluginutil.PluginVaultVersionEnv)
+	pluginVersion := os.Getenv(pluginutil.PluginVaultVersionEnv)
+
 	projectURL := "https://www.vaultproject.io/"
 	rt := runtime.Version()
-	return fmt.Sprintf("Vault/%s (+%s; %s)", version, projectURL, rt)
+
+	// assign a GUID to the user-agent used in making requests
+	// create the basic user agent string
+	ua := fmt.Sprintf("Vault/%s (+%s; %s)", pluginVersion, projectURL, rt)
```

Comment:
How come we're setting the user agent in here slightly differently (i.e. not calling `useragent.String()` from `sdk/helper/useragent`)? `pluginVersion` in here can be an empty string if ran as a builtin plugin. Is there a particular reason we're constructing the string manually vs calling the helpers on the `useragent` package similar to:

https://github.com/hashicorp/vault-plugin-secrets-azure/blob/8e44df36e43f6e0bfb99e0ffa16d5b90e1150127/provider.go#L83-L88

Reply:
I've updated this to use `sdk/helper/useragent`